File: 019. Insertion Sort/InsertionSort.java
import java.util.*;

class InsertionSort {
    public static int[] insertionSort(int[] array) {
        if (array.length == 0)
            return array;
        for (int i = 1; i < array.length; i++) {
            for (int j = i - 1; j >= 0; j--) {
                if (array[j + 1] < array[j]) {
                    swap(j, j + 1, array);
                }
            }
        }
        return array;
    }

    public static void swap(int i, int j, int[] array) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
}

File: 019. Insertion Sort/InsertionSort.kt
package com.algoexpert.program

fun InsertionSort(array: MutableList<Int>): List<Int> {
    if (array.size == 0) return array
    for (i in 1 until array.size) {
        for (j in i - 1 downTo 0) {
            if (array[j] > array[j + 1]) {
                swap(j, j + 1, array)
            }
        }
    }
    return array
}

fun swap(i: Int, j: Int, array: MutableList<Int>) {
    array[i] = array[j].also { array[j] = array[i] }
}
